scripts/apothecary/apothecary

```diff
@@ -583,7 +584,43 @@ function downloadFormula(){
 	cd $BUILD_DIR
 
 	if [ -e $1 ] ; then
-		echo "... skipping, src dir already exists"
+		#we have to remove these when we build with different architectures in vs
+		if [ "$1" == "glfw" ] ; then
+			rm -r glfw
+			if [ ! -z "$USE_GIT" ] && [[ "${GIT_URL:+x}" ]] ; then
+				gitclone $1 $GIT_URL ${GIT_TAG:-master}
+			else
+				download
+			fi
+			if [ ! -e $1 ] ; then
+				echoError " It looks like downloading failed for \"$1\""
+				exit 1
+			fi
+		elif [ "$1" == "rtAudio" ] ; then
+			rm -r rtAudio
+			if [ ! -z "$USE_GIT" ] && [[ "${GIT_URL:+x}" ]] ; then
+				gitclone $1 $GIT_URL ${GIT_TAG:-master}
+			else
+				download
+			fi
+			if [ ! -e $1 ] ; then
+				echoError " It looks like downloading failed for \"$1\""
+				exit 1
+			fi
+		elif [ "$1" == "tess2" ] ; then
+			rm -r tess2
+			if [ ! -z "$USE_GIT" ] && [[ "${GIT_URL:+x}" ]] ; then
+				gitclone $1 $GIT_URL ${GIT_TAG:-master}
+			else
+				download
+			fi
+			if [ ! -e $1 ] ; then
+				echoError " It looks like downloading failed for \"$1\""
+				exit 1
+			fi
+		else
+			echo "... skipping, src dir already exists"
+		fi
```

Comment:
yes i think in general it would be a good practice to delete and uncompress or clone everything every time. if we patch things for one platform or there are any thing remaining from a previous build it could be problematic.

also with git you do `git clone -depth=1 ...` and it'll only get the files as they are on the last commit instead of the whole history

we could add some flag so it doesn't delete the previous download if you know what you are doing but by now i would just move this code to the respective formulas